src/encoded/types/quality_metric.py

```diff
@@ -103,6 +152,71 @@ class QualityMetricBamqc(QualityMetric):
     schema = load_schema('encoded:schemas/quality_metric_bamqc.json')
     embedded_list = QualityMetric.embedded_list
 
+    @calculated_property(schema=QC_SUMMARY_SCHEMA)
+    def quality_metric_summary(self, request):
+        qc = self.properties
+        qc_summary = []
+        
+        if 'Total Reads' not in qc:
+            return
+
+        def total_number_of_reads_per_type(qc):
+            unmapped = 0
+            multimapped = 0
+            duplicates = 0
+            walks = 0
+
+            for key in qc.keys():
+                if "N" in key or "X" in key:
+                    unmapped = unmapped + qc.get(key)
+                elif "M" in key and key != "NM":
+                    multimapped = multimapped + qc.get(key)
+                elif "DD" in key:
+                    duplicates = qc.get(key)
+                elif "WW" in key:
+                    walks = qc.get(key)
+
+            return unmapped, multimapped, duplicates, walks
+
+        unmapped_reads, multi_reads, duplicates, walks = total_number_of_reads_per_type(
+            qc)
+
+        def percent_of_reads(numVal):
+            '''convert to percentage of Total reads in bam file'''
+            return round((int(numVal) / int(qc.get('Total Reads'))) * 100 * 1000) / 1000
+
+        def million(numVal):
+            return str(round(int(numVal) / 10000) / 100) + "m"
+
+        def tooltip(numVal):
+            return "Percent of total reads (=%s)" % million(int(numVal))
```

Comment:
You should refactor these into a utility file (or at top level in this file) since they are used more than once.

Reply:
We'll keep this implementation as is since only `million` method is identical. `tooltip` and `percent_of_reads` varies depending on QC Item type.